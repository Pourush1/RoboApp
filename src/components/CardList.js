import React from "react";
import Card from "./Card";

const CardList = ({ robots }) => {
  // const cardComponent =
  return (
    <div>
      {robots.map((item, index) => {
        return (
          <Card
            key={index}
            id={item.id}
            name={item.name}
            username={item.username}
            email={item.email}
          />
        );
      })}
    </div>
  );
};

export default CardList;
