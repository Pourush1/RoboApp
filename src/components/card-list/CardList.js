import React, { Component } from "react";
import PropTypes from "prop-types";
import Card from "../card/Card";

//const CardList = ({ robots }) => {

class CardList extends Component {
  render() {
    const { robots } = this.props;
    //console.log(robots);
    return (
      <div>
        {robots.map((item, index) => {
          return (
            <Card
              key={index}
              id={item.id}
              name={item.name}
              username={item.username}
              email={item.email}
            />
          );
        })}
      </div>
    );
  }
}

CardList.propTypes = {
  robotsArray: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      username: PropTypes.string,
      email: PropTypes.string
    })
  )
};

export default CardList;
