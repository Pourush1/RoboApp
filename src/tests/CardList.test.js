import React from "react";
import { shallow } from "enzyme";
import CardList from "../components/CardList";
import checkPropTypes from "check-prop-types";

const filteredRobots = [
  {
    id: 1,
    name: "Leanne Graham",
    username: "Bret",
    email: "Sincere@april.biz"
  }
];

describe("CardList Component", () => {
  //SnapShot Testing
  describe("Doing Snapshot Testing", () => {
    test("renders without crashing", () => {
      expect(shallow(<CardList robots={filteredRobots} />)).toMatchSnapshot();
    });
  });

  // Props Testing
  describe("Props Testing", () => {
    test("when provided with an array of Cards", () => {
      const cards = [
        {
          id: 1,
          name: "Leanne Graham",
          username: "Bret",
          email: "Sincere@april.biz"
        },
        {
          id: 2,
          name: "Ervin Howell",
          username: "Antonette",
          email: "Shanna@melissa.tv"
        }
      ];
      const CardListInstance = shallow(<CardList robots={cards} />);

      CardListInstance.find("Card").forEach(cardInstance => {
        const cardProps = cardInstance.props();
        const matchingCards = cards.find(card => card.id === cardProps.id);
        expect(cardProps.name).toBe(matchingCards.name);
      });
    });
  });

  // Checking prototypes
  describe("Checking PropTypes", () => {
    test("Should not throw a warning", () => {
      const expectedProps = {
        robotsArray: [
          {
            id: 1,
            name: "Leanne Graham",
            username: "Bret",
            email: "Sincere@april.biz"
          }
        ]
      };
      const propsErr = checkPropTypes(
        CardList.propTypes,
        expectedProps,
        "props",
        CardList.name
      );
      expect(propsErr).toBeUndefined();
    });
  });
});
