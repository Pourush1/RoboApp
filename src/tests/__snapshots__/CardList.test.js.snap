// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Component Doing Snapshot Testing renders without crashing 1`] = `
<div>
  <Card
    email="Sincere@april.biz"
    id={1}
    key="0"
    name="Leanne Graham"
    username="Bret"
  />
</div>
`;
